log/with_logger.go

```diff
@@ -81,7 +81,7 @@ func (l *withLogger) Error(msg string, keyvals ...interface{}) {
 
 func (l *withLogger) WithCallerSkip(depth int) Logger {
 	if sl, ok := l.logger.(WithSkipCallers); ok {
-		return newWithLogger(sl.WithCallerSkip(depth), l.keyvals...)
+		return newWithLogger(sl.WithCallerSkip(depth-1), l.keyvals...)
```

Comment:
I'm sorry, this doesn't quite make sense. This output
```
--- skip: 0
--- skip: 1
--- skip: -1
--- skip: 1
--- skip: 0
--- skip: 1
```
Is not expected, a logger should never see a negative depth.

Reply:
Yeah I was unsure about that as well. Also, there is a double skipping here and inside newWithLogger. If combined (with a bit of refactoring of `newWithLogger`), the logger will never see a negative depth.

Looks like it's assumed that `WithSkipCallers` is meant to be always implemented with `WithLogger`, so this scenario is not applicable.